src/Common/src/Interop/Windows/HttpApi/Interop.HttpApi.cs

```diff
@@ -0,0 +1,1017 @@
+
+        [DllImport(Libraries.HttpApi, SetLastError = true)]
+        internal static extern uint HttpSendHttpResponse(SafeHandle requestQueueHandle, ulong requestId, uint flags, HTTP_RESPONSE* pHttpResponse, void* pCachePolicy, uint* pBytesSent, SafeLocalAllocHandle pRequestBuffer, uint requestBufferLength, NativeOverlapped* pOverlapped, void* pLogData);
+
+        [DllImport(Libraries.HttpApi, SetLastError = true)]
+        internal static extern uint HttpWaitForDisconnect(SafeHandle requestQueueHandle, ulong connectionId, NativeOverlapped* pOverlapped);
+
+        [DllImport(Libraries.HttpApi, SetLastError = true)]
+        internal static extern uint HttpReceiveRequestEntityBody(SafeHandle requestQueueHandle, ulong requestId, uint flags, void* pEntityBuffer, uint entityBufferLength, out uint bytesReturned, NativeOverlapped* pOverlapped);
+
+        [DllImport(Libraries.HttpApi, SetLastError = true)]
+        internal static extern uint HttpSendResponseEntityBody(SafeHandle requestQueueHandle, ulong requestId, uint flags, ushort entityChunkCount, HTTP_DATA_CHUNK* pEntityChunks, uint* pBytesSent, SafeLocalAllocHandle pRequestBuffer, uint requestBufferLength, NativeOverlapped* pOverlapped, void* pLogData);
+
+        [DllImport(Libraries.HttpApi, SetLastError = true)]
+        internal static extern unsafe uint HttpCloseRequestQueue(IntPtr pReqQueueHandle);
+
+        [DllImport(Libraries.HttpApi, SetLastError = true)]
+        internal static extern uint HttpCancelHttpRequest(SafeHandle requestQueueHandle, ulong requestId, IntPtr pOverlapped);
+
+        [DllImport(Libraries.HttpApi, SetLastError = true)]
+        internal static extern uint HttpCloseServerSession(ulong serverSessionId);
+
+        internal class SafeLocalFreeChannelBinding : ChannelBinding
+        {
+            private const int LMEM_FIXED = 0;
+            private int _size;
+
+            public override int Size
+            {
+                get { return _size; }
+            }
+
+            public static SafeLocalFreeChannelBinding LocalAlloc(int cb)
+            {
+                SafeLocalFreeChannelBinding result;
+
+                result = LocalAllocChannelBinding(LMEM_FIXED, (UIntPtr)cb);
+                if (result.IsInvalid)
+                {
+                    result.SetHandleAsInvalid();
+                    throw new OutOfMemoryException();
+                }
+
+                result._size = cb;
+                return result;
+            }
+
+            override protected bool ReleaseHandle()
+            {
+                return Interop.mincore_obsolete.LocalFree(handle) == IntPtr.Zero;
+            }
+        }
+
+        [DllImport(Interop.Libraries.Heap, EntryPoint = "LocalAlloc", SetLastError = true)]
+        internal static extern SafeLocalFreeChannelBinding LocalAllocChannelBinding(int uFlags, UIntPtr sizetdwBytes);
```

Comment:
Nit: Why not just call the method LocalAlloc rather than calling it LocalAllocChannelBinding and using EntryPoint to change its name?